### PR TITLE
Write ln(likelihood) in inference result files

### PR DESCRIPTION
**Pull request by micamu**
This pull request is to write the natural logarithm of the likelihood in the result hdf files from pycbc_inference. The likelihood can then be used to compute the Bayes factor or as a sanity check in suspicious PE runs.

### CODE REVIEW

File: pycbc/inference/sampler.py [hunk 1]
```diff
@@ -48,9 +48,16 @@ def acceptance_fraction(self):
         return ValueError("acceptance_fraction function not set.")
 
     @property
+    def lnpost(self):
+    """ This function should return the natural logarithm of the likelihood
+    as an niterations x nwalker array.
+    """
+    return ValueError("lnpost function not set.")
```
Comment (cdcapano) [inline pycbc/inference/sampler.py]: Lines 52-55 need to be indented.

File: pycbc/io/inference_hdf.py [hunk 2]
```diff
@@ -453,3 +453,63 @@ def write_acl(self, acl):
...
+                if data is not None:
+                    lnlikelihood_subset[start:end] = lnlikelihood[i,start:end]
+                group_name.create_dataset(dataset_name,
+                                         data=lnlikelihood_subset)
+
+            # write all samples in range from walker
+            else:
+                if end > len(lnlikelihood[i,:]):
+                    end = None
+                lnlikelihood_subset = lnlikelihood[i,start_end]
```
Comment (cdcapano) [inline pycbc/io/inference_hdf.py]: `start_end` isn't defined; I think you mean `start:end` here.

**Comment by micamu** [comment]
Sorry, silly Friday-afternoon typos. I will fix them now and will start a short run to test it.

**Comment by micamu** [comment]
Tested, fixed error, and worked. Code is ready to merge.

**Pull request merged**